tests/test_global.py

```diff
@@ -0,0 +1,292 @@
+from settings import nso_username, nso_password, path_geojson, output_path
+import os
+
+# Optional
+from settings import (
+    height_band_filepath,
+    cloud_detection_model_path,
+    links_must_contain,
+)
+import rasterio
+from rasterio.plot import show
+import satellite_images_nso._manipulation.nso_manipulator as nso_manipulator
+import matplotlib.pyplot as plt
+
+# the cloud_recognition.api needs to be imported from the natura2000 data science repo
+from cloud_recognition.api import detect_clouds
+import pickle
+import numpy as np
+
+
+def all_true(arr):
+    for val in arr:
+        if not val:
+            return False
+    return True
+
+
+def all_false(arr):
+    for val in arr:
+        if val:
+            return False
+    return True
+
+
+def download_zip(url):
+    # URL of the zip file you want to download
+    zip_url = url
+
+    # Local path where you want to save the downloaded zip file
+    local_tiff_path = output_path + "/" + url.split("/")[-1]
+
+    # Send a GET request to the URL to download the zip file
+    response = requests.get(zip_url)
+
+    # Check if the request was successful (status code 200)
+    if response.status_code == 200:
+        # Write the content of the response (the zip file) to a local file
+        with open(local_tiff_path, "wb") as file:
+            file.write(response.content)
+
+        print(f"zip file has been downloaded to {local_tiff_path}")
+    else:
+        print("Failed to download zip file:", response.status_code)
+
+
+georegion = nso.nso_georegion(
+    path_to_geojson=path_geojson,
+    output_folder=output_path,
+    username=nso_username,
+    password=nso_password,
+)
+links = georegion.retrieve_download_links(max_diff=0.5, start_date="2011-01-01")
+
+
+### Check if test data exists else download the data.
+
+if not os.path.exists(
+    output_path + "/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+):
+    download_zip(
+        "https://e34a505986aa74678a5a0e0f.blob.core.windows.net/satellite-images-nso/Test_regions/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+    )
+
+if not os.path.exists(
+    output_path + "/20191202_110523_SV1-04_SV_RD_11bit_RGBI_50cm.zip"
+):
+    download_zip(
+        "https://e34a505986aa74678a5a0e0f.blob.core.windows.net/satellite-images-nso/Test_regions/20230513_104139_PNEO-03_1_1_30cm_RGBNED_12bit_PNEO.zip"
+    )
+
+
+### Link tests.
+
+
+def test_found_links():
+
+    assert len(links) > 0, "No links found error!"
+
+
+def test_link_columns():
+    # Columns to check for existence
+    columns_to_check = [
+        "link",
+        "percentage_geojson",
+        "missing_polygon",
+        "covered_polygon",
+    ]
+
+    # Check if columns exist
+    existing_columns = [col for col in columns_to_check if col in links.columns]
+
+    assert len(existing_columns) == len(
+        columns_to_check
+    ), "Certain columns are not in the links"
+
+
+def test_pneo_links():
```

Comment:
Isn't this basically the same test as test_found_links, but more rigid? It seems redundant.

Reply:
Okay then.